devicetypes/krlaframboise/ecolink-chime-siren.src/ecolink-chime-siren.groovy

```diff
@@ -0,0 +1,831 @@
+	logTrace "$cmd"
+	sendEventIfNew("firmwareVersion", (cmd.applicationVersion + (cmd.applicationSubVersion / 100)))
+}
+
+
+void zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	logTrace "${cmd}"
+	sendEventIfNew("switch", (cmd.value ? "on" : "off"))
+}
+
+
+void zwaveEvent(physicalgraph.zwave.Command cmd) {
+	logDebug "Unhandled zwaveEvent: $cmd"
+}
+
+
+void zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	logTrace "${cmd}"
+
+	if (cmd.notificationType == 0x08) {
+		String powerSource = null
+		switch (cmd.event) {
+			case 0x02:
+				logDebug "AC Mains Disconnected"
+				powerSource = "battery"
+				break
+			case 0x03:
+				logDebug "AC Mains Re-Connected"
+				powerSource = "mains"
+				break
+			case 0x0C:
+				// logDebug "Battery is charging"
+				// powerSource = "mains"
+				break
+			case 0x0D:
+				// logDebug "battery is fully charged"
+				// powerSource = "mains"
+				break
+			case 0x0E:
+				// logDebug "charge battery soon"
+				// powerSource = "battery"
+				break
+			case 0x0F:
+				// logDebug "charge battery now"
+				// powerSource = "battery"
+				break
+			default:
+				logDebug "Unknown Power Event: ${cmd}"
+		}
+		if (powerSource) {
+			sendEventIfNew("powerSource", powerSource)
+		}
+	}
+	else if (cmd.notificationType == 0x09) {
+		if (cmd.event == 0x04) {
+			// 0x5501 Watchdog Reset on the Z-Wave Module
+			// 0x5502 Watchdog Reset on the STM32
+		}
+	}
+	else {
+		logDebug "Unknown notificationType: ${cmd}"
+	}
+}
+
+
+void handleBatteryReport(String description) {
+	// logTrace "handleBatteryReport(${description})..."
+
+	// BATTERY V2 NOT SUPPORTED BY SMARTTHINGS
+
+	// The handler can't rely on the Power Management Notification Reports to determine the power source because of a firmware issue that occurs when the device is plugged back in after the battery gets low.
+
+	Map cmd = parseCommand(description)
+	if (cmd?.payloadBytes?.size() >= 2) {
+
+		int value = hexToInt(cmd.payloadBytes[0])
+
+		sendEvent(getEventMap("battery", (value == 0xFF ? 1 : value), "%"))
+
+		try {
+			String powerSource = null
+
+			int chargingStatus = Integer.parseInt(Integer.toBinaryString(hexToInt(cmd.payloadBytes[1])).padLeft(8, "0").substring(0, 2), 2)
+
+			switch (chargingStatus) {
+				case 0:
+					// discharging
+					powerSource = "battery"
+					break
+
+				case 1:
+					// charging
+					powerSource = "mains"
+					break
+
+				case 2:
+					// maintaining
+					powerSource = "mains"
+					break
+			}
+
+			if (powerSource) {
+				sendEventIfNew("powerSource", powerSource)
+			}
+		}
+		catch (ex) {
+			log.warn "Unable to parse battery charging status from ${description}"
+		}
+	}
+}
+
+
+void handleSoundSwitchEvent(String description) {
+	// logTrace "handleSoundSwitchEvent(${description})..."
+
+	// SOUND SWITCH CC NOT SUPPORTED BY SMARTTHINGS
+
+	Map cmd = parseCommand(description)
+
+	switch (cmd?.command) {
+		case "7907":
```

Comment:
Please could You use constants so it would be more descriptive ?

Reply:
see new merge request